test/unit/models/flag.test.js

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable security/detect-object-injection */
+const chai = require("chai");
+const { expect } = chai;
+
+const cleanDb = require("../../utils/cleanDb");
+
+const flagModel = require("../../../models/flag");
+const firestore = require("../../../utils/firestore");
+const flagFirestore = firestore.collection("featureFlags");
+const flagData = require("../../fixtures/flag/flag")();
+
+describe("flags", function () {
+  let flagId;
+
+  beforeEach(async function () {
+    flagId = await flagModel.addFlag(flagData);
+  });
+
+  afterEach(async function () {
+    await cleanDb();
+  });
+
+  describe("add  flag", function () {
+    it("Should return flag Id", async function () {
+      const fetched = await flagFirestore.doc(flagId).get();
+      const fetchedData = fetched.data();
+      const fetchedId = fetched.id;
+      Object.keys(flagData).forEach((key) => {
+        expect(flagData[key]).to.deep.equal(fetchedData[key]);
+      });
+      expect(fetchedId).to.equal(flagId);
+    });
+  });
+
+  describe("fetch all flag", function () {
+    it("Should return all flags", async function () {
+      const fetched = await flagFirestore.doc(flagId).get();
+      const fetchedData = fetched.data();
+      Object.keys(flagData).forEach((key) => {
+        expect(flagData[key]).to.deep.equal(fetchedData[key]);
+      });
+    });
```

Comment:
You can add a test case when no flag was found or 0 flags were returned.

Reply:
I have made changes please review them